src/solvers/refinement/string_refinement.cpp

```diff
@@ -1955,50 +2013,38 @@ exprt substitute_array_lists(exprt expr, size_t string_max_length)
 /// \return an expression
 exprt string_refinementt::get(const exprt &expr) const
 {
-  const std::function<exprt(const exprt &)> super_get=[this](const exprt &expr)
-  { return (exprt) supert::get(expr); };
+  const auto super_get=[this](const exprt &expr)
+  { return supert::get(expr); };
   exprt ecopy(expr);
-  replace_expr(symbol_resolve, ecopy);
-  if(is_char_array(ns, ecopy.type()))
-  {
-    auto it_content=found_content.find(ecopy);
-    if(it_content!=found_content.end())
-      return it_content->second;
+  (void)symbol_resolve.replace_expr(ecopy);
 
-    auto it=found_length.find(ecopy);
-    if(it!=found_length.end())
-      return get_array(
-        super_get,
-        ns,
-        generator.max_string_length,
-        ecopy,
-        it->second);
-  }
-  else if(ecopy.id()==ID_struct)
+  if(ecopy.type().id()==ID_array)
   {
-    if(const auto string=expr_cast<string_exprt>(ecopy))
+    array_string_exprt &arr=to_array_string_expr(ecopy);
+    arr.length()=generator.get_length_of_string_array(arr);
+    const auto arr_model_opt=
+      get_array(super_get, ns, generator.max_string_length, debug(), arr);
+    // Should be refactored with get array or get array in model
+    if(arr_model_opt)
     {
-      const exprt &content=string->content();
-      const exprt &length=string->length();
-
-      const exprt arr=get_array(
-        super_get,
-        ns,
-        generator.max_string_length,
-        content,
-        length);
-      ecopy=string_exprt(length, arr, string->type());
+      const exprt arr_model=simplify_expr(*arr_model_opt, ns);
+      const exprt concretized_array=concretize_arrays_in_expression(
+        arr_model, generator.max_string_length, ns);
+      debug() << "get " << from_expr(ns, "", expr) << " --> "
```

Comment:
I was debugging something with Justing, and this debug output and the one below make this impossible, get is called too often. Please delete.